### DISCUSSION TITLE
How to restart and reattach to serial

### DISCUSSION THREAD

**Discussion by jjmakin** [rhinstaller/anaconda#3712] [Q&A] [answered]
Hello folks, occasionally I mess up kickstart files for automated setups and the process cannot complete, however it's super useful being  able to salvage the build, ssh in to the box, fix the typo and start again without having to do a whole hardware reboot

systemctl stop anaconda.service
anaconda-cleanup
systemctl start --no-block anaconda.service

This is great via an SSH connection, but as I watch a number of builds over serial (ipmi or virsh console), is it possible somehow to restart anaconda and have it redirect back to the initial ttyS0 or ttyS1?

**Comment by jkonecny12** [comment]
Hi, definitely an interesting use-case but I fear that Anaconda wasn't designed for this. I guess we could do some changes in the code to make the existing redirect to ttySX but I guess there are far more complications you could come across.

However, to help you with this. What is the problem there? That you don't see the Anaconda anymore on the existing link after the service is started? If yes, than the problem is probably just a detached tmux session. In that case just run
```
$ tmux ls                                  # you should see the detached tmux instance 
$ tmux attach-session -t <id>  # to attach the tmux instance
```

**Comment by jjmakin** [comment, marked as answer]
Yes if I ditch my serial over lan session and SSH into anaconda I can join the new tmux instance.
This might not be necessary anymore. On centos8 build I have found I can simply call systemctl restart anaconda and the output still takes place on ttySX
